Engine/Component/ComponentImage.cpp

```diff
@@ -60,12 +60,29 @@ void ComponentImage::Render(float4x4* projection)
 
 	if (texture_to_render != nullptr)
 	{
-		float4x4* model = &owner->transform_2d.GetSizedGlobalModelMatrix();
+		ScaleOp aspect_ratio_scaling;
+
+		if (preserve_aspect_ratio)
+		{
+			if (owner->transform_2d.size.x / texture_aspect_ratio > owner->transform_2d.size.y)
+			{
+				aspect_ratio_scaling = float4x4::Scale(float3(owner->transform_2d.size.y * texture_aspect_ratio, owner->transform_2d.size.y, 1.f));
+			}
+			else
+			{
+				aspect_ratio_scaling = float4x4::Scale(float3(owner->transform_2d.size.x, owner->transform_2d.size.x / texture_aspect_ratio, 1.f));
+			}
+		}
+		else
+		{
+			aspect_ratio_scaling = float4x4::Scale(float3(owner->transform_2d.size, 1.f));
+		}
```

Comment:
Maybe encapsulate this, for ex: GetAspectRatio or something like that, and keep it out from Render.